cmd/qspi/src/lib.rs

```diff
@@ -852,7 +857,13 @@ fn qspi(context: &mut ExecutionContext) -> Result<()> {
         // We are going to hash the contents to find the differences, and
         // then erase/flash the different sectors.
         //
-        let mut address = 0u32;
+        let base_addr = subargs.addr.unwrap_or(0) as u32;
+        if base_addr % SECTOR_SIZE != 0 {
+            bail!(
+                "base address {base_addr:#x} must be divisible by sector size \
+                 ({SECTOR_SIZE})"
+            );
```

Comment:
super nitpicky, feel free to ignore: would it maybe be helpful to explicitly indicate here that the base address came from the `--addr` argument? since, if it wasn't provided by the CLI, it will be 0, which will always be well aligned, it might be useful-ish to tell the user where the misaligned address came from?

Reply:
Done!